// ngx-three/esm2020/lib/directives/ref-by-id.directive.mjs
import { Directive, Input, Self, SkipSelf } from '@angular/core';
import { createLazyObject3DProxy, isLazyObject3dProxy } from '../loaders/LazyObject3dProxy';
import * as i0 from "@angular/core";
import * as i1 from "../generated";
export class RefByIdDirective {
    constructor(host, parent) {
        this.host = host;
        this.parent = parent;
        this.host.autoAddToParent = false;
        this.host.autoDispose = false;
        this.host.objRef = createLazyObject3DProxy();
    }
    set refById(id) {
        this.id = id;
        this.subscription?.unsubscribe();
        this.subscription = this.parent.objRef$.subscribe((obj) => {
            this.parentObj = obj;
            this.applyRef();
        });
    }
    applyRef() {
        if (!this.parentObj || this.id === undefined) {
            return;
        }
        const ref = this.findById(this.parentObj, this.id);
        if (ref) {
            this.setRef(this.host, ref);
        }
    }
    findById(parentObj, id) {
        if ((parentObj.name ?? parentObj.uuid) === id) {
            return parentObj;
        }
        for (const child of parentObj.children) {
            const object = this.findById(child, id);
            if (object !== undefined) {
                return object;
            }
        }
        return undefined;
    }
    setRef(host, ref) {
        const objRef = host.objRef;
        if (!objRef) {
            return;
        }
        if (isLazyObject3dProxy(objRef)) {
            objRef.objRef = ref;
            host.objRef = objRef;
        }
    }
    ngOnDestroy() {
        this.subscription?.unsubscribe();
    }
}
RefByIdDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RefByIdDirective, deps: [{ token: i1.ThObject3D, self: true }, { token: i1.ThObject3D, skipSelf: true }], target: i0.ɵɵFactoryTarget.Directive });
RefByIdDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: RefByIdDirective, selector: "[refById]", inputs: { refById: "refById" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RefByIdDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[refById]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D, decorators: [{
                    type: Self
                }] }, { type: i1.ThObject3D, decorators: [{
                    type: SkipSelf
                }] }]; }, propDecorators: { refById: [{
                type: Input
            }] } });
