// ngx-three/esm2020/lib/generated/ThArcballControls.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { ArcballControls } from 'three/examples/jsm/controls/ArcballControls';
import { ThControlBase } from '../ThControlBase';
import { applyValue } from '../util';
import * as i0 from "@angular/core";
export class ThArcballControls extends ThControlBase {
    getType() {
        return ArcballControls;
    }
    set camera(value) {
        if (this._objRef) {
            this._objRef.camera = value;
        }
    }
    // @ts-ignore
    get camera() {
        return this._objRef?.camera;
    }
    set domElement(value) {
        if (this._objRef) {
            this._objRef.domElement = value;
        }
    }
    // @ts-ignore
    get domElement() {
        return this._objRef?.domElement;
    }
    set scene(value) {
        if (this._objRef) {
            this._objRef.scene = value;
        }
    }
    // @ts-ignore
    get scene() {
        return this._objRef?.scene;
    }
    set focusAnimationTime(value) {
        if (this._objRef) {
            this._objRef.focusAnimationTime = value;
        }
    }
    // @ts-ignore
    get focusAnimationTime() {
        return this._objRef?.focusAnimationTime;
    }
    set enabled(value) {
        if (this._objRef) {
            this._objRef.enabled = value;
        }
    }
    // @ts-ignore
    get enabled() {
        return this._objRef?.enabled;
    }
    set enablePan(value) {
        if (this._objRef) {
            this._objRef.enablePan = value;
        }
    }
    // @ts-ignore
    get enablePan() {
        return this._objRef?.enablePan;
    }
    set enableRotate(value) {
        if (this._objRef) {
            this._objRef.enableRotate = value;
        }
    }
    // @ts-ignore
    get enableRotate() {
        return this._objRef?.enableRotate;
    }
    set enableZoom(value) {
        if (this._objRef) {
            this._objRef.enableZoom = value;
        }
    }
    // @ts-ignore
    get enableZoom() {
        return this._objRef?.enableZoom;
    }
    set enableGizmos(value) {
        if (this._objRef) {
            this._objRef.enableGizmos = value;
        }
    }
    // @ts-ignore
    get enableGizmos() {
        return this._objRef?.enableGizmos;
    }
    set adjustNearFar(value) {
        if (this._objRef) {
            this._objRef.adjustNearFar = value;
        }
    }
    // @ts-ignore
    get adjustNearFar() {
        return this._objRef?.adjustNearFar;
    }
    set scaleFactor(value) {
        if (this._objRef) {
            this._objRef.scaleFactor = value;
        }
    }
    // @ts-ignore
    get scaleFactor() {
        return this._objRef?.scaleFactor;
    }
    set dampingFactor(value) {
        if (this._objRef) {
            this._objRef.dampingFactor = value;
        }
    }
    // @ts-ignore
    get dampingFactor() {
        return this._objRef?.dampingFactor;
    }
    set wMax(value) {
        if (this._objRef) {
            this._objRef.wMax = value;
        }
    }
    // @ts-ignore
    get wMax() {
        return this._objRef?.wMax;
    }
    set enableAnimations(value) {
        if (this._objRef) {
            this._objRef.enableAnimations = value;
        }
    }
    // @ts-ignore
    get enableAnimations() {
        return this._objRef?.enableAnimations;
    }
    set enableGrid(value) {
        if (this._objRef) {
            this._objRef.enableGrid = value;
        }
    }
    // @ts-ignore
    get enableGrid() {
        return this._objRef?.enableGrid;
    }
    set cursorZoom(value) {
        if (this._objRef) {
            this._objRef.cursorZoom = value;
        }
    }
    // @ts-ignore
    get cursorZoom() {
        return this._objRef?.cursorZoom;
    }
    set minFov(value) {
        if (this._objRef) {
            this._objRef.minFov = value;
        }
    }
    // @ts-ignore
    get minFov() {
        return this._objRef?.minFov;
    }
    set maxFov(value) {
        if (this._objRef) {
            this._objRef.maxFov = value;
        }
    }
    // @ts-ignore
    get maxFov() {
        return this._objRef?.maxFov;
    }
    set minDistance(value) {
        if (this._objRef) {
            this._objRef.minDistance = value;
        }
    }
    // @ts-ignore
    get minDistance() {
        return this._objRef?.minDistance;
    }
    set maxDistance(value) {
        if (this._objRef) {
            this._objRef.maxDistance = value;
        }
    }
    // @ts-ignore
    get maxDistance() {
        return this._objRef?.maxDistance;
    }
    set minZoom(value) {
        if (this._objRef) {
            this._objRef.minZoom = value;
        }
    }
    // @ts-ignore
    get minZoom() {
        return this._objRef?.minZoom;
    }
    set maxZoom(value) {
        if (this._objRef) {
            this._objRef.maxZoom = value;
        }
    }
    // @ts-ignore
    get maxZoom() {
        return this._objRef?.maxZoom;
    }
    set target(value) {
        if (this._objRef) {
            this._objRef.target = applyValue(this._objRef.target, value);
        }
    }
    // @ts-ignore
    get target() {
        return this._objRef?.target;
    }
    set radiusFactor(value) {
        if (this._objRef) {
            this._objRef.radiusFactor = value;
        }
    }
    // @ts-ignore
    get radiusFactor() {
        return this._objRef?.radiusFactor;
    }
}
ThArcballControls.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThArcballControls, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThArcballControls.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThArcballControls, selector: "th-arcballControls", inputs: { camera: "camera", domElement: "domElement", scene: "scene", focusAnimationTime: "focusAnimationTime", enabled: "enabled", enablePan: "enablePan", enableRotate: "enableRotate", enableZoom: "enableZoom", enableGizmos: "enableGizmos", adjustNearFar: "adjustNearFar", scaleFactor: "scaleFactor", dampingFactor: "dampingFactor", wMax: "wMax", enableAnimations: "enableAnimations", enableGrid: "enableGrid", cursorZoom: "cursorZoom", minFov: "minFov", maxFov: "maxFov", minDistance: "minDistance", maxDistance: "maxDistance", minZoom: "minZoom", maxZoom: "maxZoom", target: "target", radiusFactor: "radiusFactor" }, providers: [
        {
            provide: ThControlBase,
            useExisting: forwardRef(() => ThArcballControls)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThArcballControls, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-arcballControls',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThControlBase,
                            useExisting: forwardRef(() => ThArcballControls)
                        }
                    ]
                }]
        }], propDecorators: { camera: [{
                type: Input
            }], domElement: [{
                type: Input
            }], scene: [{
                type: Input
            }], focusAnimationTime: [{
                type: Input
            }], enabled: [{
                type: Input
            }], enablePan: [{
                type: Input
            }], enableRotate: [{
                type: Input
            }], enableZoom: [{
                type: Input
            }], enableGizmos: [{
                type: Input
            }], adjustNearFar: [{
                type: Input
            }], scaleFactor: [{
                type: Input
            }], dampingFactor: [{
                type: Input
            }], wMax: [{
                type: Input
            }], enableAnimations: [{
                type: Input
            }], enableGrid: [{
                type: Input
            }], cursorZoom: [{
                type: Input
            }], minFov: [{
                type: Input
            }], maxFov: [{
                type: Input
            }], minDistance: [{
                type: Input
            }], maxDistance: [{
                type: Input
            }], minZoom: [{
                type: Input
            }], maxZoom: [{
                type: Input
            }], target: [{
                type: Input
            }], radiusFactor: [{
                type: Input
            }] } });
