// ngx-three/esm2020/lib/generated/ThArrowHelper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { ArrowHelper } from 'three';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThArrowHelper extends ThObject3D {
    getType() {
        return ArrowHelper;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set line(value) {
        if (this._objRef) {
            this._objRef.line = value;
        }
    }
    // @ts-ignore
    get line() {
        return this._objRef?.line;
    }
    set cone(value) {
        if (this._objRef) {
            this._objRef.cone = value;
        }
    }
    // @ts-ignore
    get cone() {
        return this._objRef?.cone;
    }
}
ThArrowHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThArrowHelper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThArrowHelper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThArrowHelper, selector: "th-arrowHelper", inputs: { line: "line", cone: "cone" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThArrowHelper) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThArrowHelper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-arrowHelper',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThArrowHelper) }]
                }]
        }], propDecorators: { line: [{
                type: Input
            }], cone: [{
                type: Input
            }] } });
