// ngx-three/esm2020/lib/generated/ThAudioListener.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { AudioListener } from 'three';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThAudioListener extends ThObject3D {
    getType() {
        return AudioListener;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set context(value) {
        if (this._objRef) {
            this._objRef.context = value;
        }
    }
    // @ts-ignore
    get context() {
        return this._objRef?.context;
    }
    set gain(value) {
        if (this._objRef) {
            this._objRef.gain = value;
        }
    }
    // @ts-ignore
    get gain() {
        return this._objRef?.gain;
    }
    set filter(value) {
        if (this._objRef) {
            this._objRef.filter = value;
        }
    }
    // @ts-ignore
    get filter() {
        return this._objRef?.filter;
    }
    set timeDelta(value) {
        if (this._objRef) {
            this._objRef.timeDelta = value;
        }
    }
    // @ts-ignore
    get timeDelta() {
        return this._objRef?.timeDelta;
    }
}
ThAudioListener.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAudioListener, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThAudioListener.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThAudioListener, selector: "th-audioListener", inputs: { context: "context", gain: "gain", filter: "filter", timeDelta: "timeDelta" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThAudioListener) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAudioListener, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-audioListener',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThAudioListener) }]
                }]
        }], propDecorators: { context: [{
                type: Input
            }], gain: [{
                type: Input
            }], filter: [{
                type: Input
            }], timeDelta: [{
                type: Input
            }] } });
