// ngx-three/esm2020/lib/generated/ThAxesHelper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { AxesHelper } from 'three';
import { ThLineSegments } from './ThLineSegments';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThAxesHelper extends ThLineSegments {
    getType() {
        return AxesHelper;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
}
ThAxesHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAxesHelper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThAxesHelper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThAxesHelper, selector: "th-axesHelper", providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThAxesHelper) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAxesHelper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-axesHelper',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThAxesHelper) }]
                }]
        }] });
