// ngx-three/esm2020/lib/generated/ThBoxLineGeometry.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { BoxLineGeometry } from 'three/examples/jsm/geometries/BoxLineGeometry';
import { ThBufferGeometry } from './ThBufferGeometry';
import * as i0 from "@angular/core";
export class ThBoxLineGeometry extends ThBufferGeometry {
    getType() {
        return BoxLineGeometry;
    }
}
ThBoxLineGeometry.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThBoxLineGeometry, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThBoxLineGeometry.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThBoxLineGeometry, selector: "th-boxLineGeometry", providers: [
        {
            provide: ThBufferGeometry,
            useExisting: forwardRef(() => ThBoxLineGeometry)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThBoxLineGeometry, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-boxLineGeometry',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThBufferGeometry,
                            useExisting: forwardRef(() => ThBoxLineGeometry)
                        }
                    ]
                }]
        }] });
