// ngx-three/esm2020/lib/generated/ThBone.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { Bone } from 'three';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThBone extends ThObject3D {
    getType() {
        return Bone;
    }
    // @ts-ignore
    get isBone() {
        return this._objRef?.isBone;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
}
ThBone.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThBone, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThBone.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThBone, selector: "th-bone", providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThBone) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThBone, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-bone',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThBone) }]
                }]
        }] });
