// ngx-three/esm2020/lib/generated/ThBox3Helper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { Box3Helper } from 'three';
import { applyValue } from '../util';
import { ThLineSegments } from './ThLineSegments';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThBox3Helper extends ThLineSegments {
    getType() {
        return Box3Helper;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set box(value) {
        if (this._objRef) {
            this._objRef.box = applyValue(this._objRef.box, value);
        }
    }
    // @ts-ignore
    get box() {
        return this._objRef?.box;
    }
}
ThBox3Helper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThBox3Helper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThBox3Helper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThBox3Helper, selector: "th-box3Helper", inputs: { box: "box" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThBox3Helper) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThBox3Helper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-box3Helper',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThBox3Helper) }]
                }]
        }], propDecorators: { box: [{
                type: Input
            }] } });
