// ngx-three/esm2020/lib/generated/ThCanvasTexture.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { CanvasTexture } from 'three';
import { ThTextureBase } from '../ThTextureBase';
import { ThTexture } from './ThTexture';
import * as i0 from "@angular/core";
export class ThCanvasTexture extends ThTexture {
    getType() {
        return CanvasTexture;
    }
    // @ts-ignore
    get isCanvasTexture() {
        return this._objRef?.isCanvasTexture;
    }
}
ThCanvasTexture.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCanvasTexture, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThCanvasTexture.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThCanvasTexture, selector: "th-canvasTexture", providers: [{ provide: ThTextureBase, useExisting: forwardRef(() => ThCanvasTexture) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCanvasTexture, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-canvasTexture',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThTextureBase, useExisting: forwardRef(() => ThCanvasTexture) }]
                }]
        }] });
