// ngx-three/esm2020/lib/generated/ThCompressedArrayTexture.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { CompressedArrayTexture } from 'three';
import { ThTextureBase } from '../ThTextureBase';
import { ThCompressedTexture } from './ThCompressedTexture';
import * as i0 from "@angular/core";
export class ThCompressedArrayTexture extends ThCompressedTexture {
    getType() {
        return CompressedArrayTexture;
    }
    // @ts-ignore
    get isCompressedArrayTexture() {
        return this._objRef?.isCompressedArrayTexture;
    }
    set wrapR(value) {
        if (this._objRef) {
            this._objRef.wrapR = value;
        }
    }
    // @ts-ignore
    get wrapR() {
        return this._objRef?.wrapR;
    }
}
ThCompressedArrayTexture.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCompressedArrayTexture, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThCompressedArrayTexture.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThCompressedArrayTexture, selector: "th-compressedArrayTexture", inputs: { wrapR: "wrapR" }, providers: [
        {
            provide: ThTextureBase,
            useExisting: forwardRef(() => ThCompressedArrayTexture)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCompressedArrayTexture, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-compressedArrayTexture',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThTextureBase,
                            useExisting: forwardRef(() => ThCompressedArrayTexture)
                        }
                    ]
                }]
        }], propDecorators: { wrapR: [{
                type: Input
            }] } });
