// ngx-three/esm2020/lib/generated/ThCompressedTexture.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { CompressedTexture } from 'three';
import { ThTextureBase } from '../ThTextureBase';
import { ThTexture } from './ThTexture';
import * as i0 from "@angular/core";
export class ThCompressedTexture extends ThTexture {
    getType() {
        return CompressedTexture;
    }
    // @ts-ignore
    get isCompressedTexture() {
        return this._objRef?.isCompressedTexture;
    }
    set mipmaps(value) {
        if (this._objRef) {
            this._objRef.mipmaps = value;
        }
    }
    // @ts-ignore
    get mipmaps() {
        return this._objRef?.mipmaps;
    }
    set format(value) {
        if (this._objRef) {
            this._objRef.format = value;
        }
    }
    // @ts-ignore
    get format() {
        return this._objRef?.format;
    }
    set flipY(value) {
        if (this._objRef) {
            this._objRef.flipY = value;
        }
    }
    // @ts-ignore
    get flipY() {
        return this._objRef?.flipY;
    }
    set generateMipmaps(value) {
        if (this._objRef) {
            this._objRef.generateMipmaps = value;
        }
    }
    // @ts-ignore
    get generateMipmaps() {
        return this._objRef?.generateMipmaps;
    }
}
ThCompressedTexture.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCompressedTexture, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThCompressedTexture.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThCompressedTexture, selector: "th-compressedTexture", inputs: { mipmaps: "mipmaps", format: "format", flipY: "flipY", generateMipmaps: "generateMipmaps" }, providers: [
        {
            provide: ThTextureBase,
            useExisting: forwardRef(() => ThCompressedTexture)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCompressedTexture, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-compressedTexture',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThTextureBase,
                            useExisting: forwardRef(() => ThCompressedTexture)
                        }
                    ]
                }]
        }], propDecorators: { mipmaps: [{
                type: Input
            }], format: [{
                type: Input
            }], flipY: [{
                type: Input
            }], generateMipmaps: [{
                type: Input
            }] } });
