// ngx-three/esm2020/lib/generated/ThConvexGeometry.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { ConvexGeometry } from 'three/examples/jsm/geometries/ConvexGeometry';
import { ThBufferGeometry } from './ThBufferGeometry';
import * as i0 from "@angular/core";
export class ThConvexGeometry extends ThBufferGeometry {
    getType() {
        return ConvexGeometry;
    }
}
ThConvexGeometry.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThConvexGeometry, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThConvexGeometry.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThConvexGeometry, selector: "th-convexGeometry", providers: [
        {
            provide: ThBufferGeometry,
            useExisting: forwardRef(() => ThConvexGeometry)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThConvexGeometry, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-convexGeometry',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThBufferGeometry,
                            useExisting: forwardRef(() => ThConvexGeometry)
                        }
                    ]
                }]
        }] });
