// ngx-three/esm2020/lib/generated/ThCubeTexture.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { CubeTexture } from 'three';
import { ThTextureBase } from '../ThTextureBase';
import { ThTexture } from './ThTexture';
import * as i0 from "@angular/core";
export class ThCubeTexture extends ThTexture {
    getType() {
        return CubeTexture;
    }
    // @ts-ignore
    get isCubeTexture() {
        return this._objRef?.isCubeTexture;
    }
    set mapping(value) {
        if (this._objRef) {
            this._objRef.mapping = value;
        }
    }
    // @ts-ignore
    get mapping() {
        return this._objRef?.mapping;
    }
    set flipY(value) {
        if (this._objRef) {
            this._objRef.flipY = value;
        }
    }
    // @ts-ignore
    get flipY() {
        return this._objRef?.flipY;
    }
}
ThCubeTexture.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTexture, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThCubeTexture.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThCubeTexture, selector: "th-cubeTexture", inputs: { mapping: "mapping", flipY: "flipY" }, providers: [{ provide: ThTextureBase, useExisting: forwardRef(() => ThCubeTexture) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTexture, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-cubeTexture',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThTextureBase, useExisting: forwardRef(() => ThCubeTexture) }]
                }]
        }], propDecorators: { mapping: [{
                type: Input
            }], flipY: [{
                type: Input
            }] } });
