// ngx-three/esm2020/lib/generated/ThDecalGeometry.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { DecalGeometry } from 'three/examples/jsm/geometries/DecalGeometry';
import { ThBufferGeometry } from './ThBufferGeometry';
import * as i0 from "@angular/core";
export class ThDecalGeometry extends ThBufferGeometry {
    getType() {
        return DecalGeometry;
    }
}
ThDecalGeometry.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDecalGeometry, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThDecalGeometry.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThDecalGeometry, selector: "th-decalGeometry", providers: [
        {
            provide: ThBufferGeometry,
            useExisting: forwardRef(() => ThDecalGeometry)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDecalGeometry, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-decalGeometry',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThBufferGeometry,
                            useExisting: forwardRef(() => ThDecalGeometry)
                        }
                    ]
                }]
        }] });
