// ngx-three/esm2020/lib/generated/ThDirectionalLightHelper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { DirectionalLightHelper } from 'three';
import { applyValue } from '../util';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThDirectionalLightHelper extends ThObject3D {
    getType() {
        return DirectionalLightHelper;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set lightPlane(value) {
        if (this._objRef) {
            this._objRef.lightPlane = value;
        }
    }
    // @ts-ignore
    get lightPlane() {
        return this._objRef?.lightPlane;
    }
    set light(value) {
        if (this._objRef) {
            this._objRef.light = value;
        }
    }
    // @ts-ignore
    get light() {
        return this._objRef?.light;
    }
    set matrix(value) {
        if (this._objRef) {
            this._objRef.matrix = applyValue(this._objRef.matrix, value);
        }
    }
    // @ts-ignore
    get matrix() {
        return this._objRef?.matrix;
    }
    set matrixAutoUpdate(value) {
        if (this._objRef) {
            this._objRef.matrixAutoUpdate = value;
        }
    }
    // @ts-ignore
    get matrixAutoUpdate() {
        return this._objRef?.matrixAutoUpdate;
    }
    set color(value) {
        if (this._objRef) {
            this._objRef.color = value;
        }
    }
    // @ts-ignore
    get color() {
        return this._objRef?.color;
    }
    set targetLine(value) {
        if (this._objRef) {
            this._objRef.targetLine = value;
        }
    }
    // @ts-ignore
    get targetLine() {
        return this._objRef?.targetLine;
    }
}
ThDirectionalLightHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDirectionalLightHelper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThDirectionalLightHelper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThDirectionalLightHelper, selector: "th-directionalLightHelper", inputs: { lightPlane: "lightPlane", light: "light", matrix: "matrix", matrixAutoUpdate: "matrixAutoUpdate", color: "color", targetLine: "targetLine" }, providers: [
        {
            provide: ThObject3D,
            useExisting: forwardRef(() => ThDirectionalLightHelper)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDirectionalLightHelper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-directionalLightHelper',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThObject3D,
                            useExisting: forwardRef(() => ThDirectionalLightHelper)
                        }
                    ]
                }]
        }], propDecorators: { lightPlane: [{
                type: Input
            }], light: [{
                type: Input
            }], matrix: [{
                type: Input
            }], matrixAutoUpdate: [{
                type: Input
            }], color: [{
                type: Input
            }], targetLine: [{
                type: Input
            }] } });
