// ngx-three/esm2020/lib/generated/ThDirectionalLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { DirectionalLight } from 'three';
import { applyValue } from '../util';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThDirectionalLight extends ThLight {
    getType() {
        return DirectionalLight;
    }
    // @ts-ignore
    get isDirectionalLight() {
        return this._objRef?.isDirectionalLight;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set castShadow(value) {
        if (this._objRef) {
            this._objRef.castShadow = value;
        }
    }
    // @ts-ignore
    get castShadow() {
        return this._objRef?.castShadow;
    }
    set position(value) {
        if (this._objRef) {
            applyValue(this._objRef.position, value);
        }
    }
    // @ts-ignore
    get position() {
        return this._objRef?.position;
    }
    set shadow(value) {
        if (this._objRef) {
            this._objRef.shadow = value;
        }
    }
    // @ts-ignore
    get shadow() {
        return this._objRef?.shadow;
    }
    set target(value) {
        if (this._objRef) {
            this._objRef.target = value;
        }
    }
    // @ts-ignore
    get target() {
        return this._objRef?.target;
    }
}
ThDirectionalLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDirectionalLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThDirectionalLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThDirectionalLight, selector: "th-directionalLight", inputs: { castShadow: "castShadow", position: "position", shadow: "shadow", target: "target" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThDirectionalLight) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDirectionalLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-directionalLight',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThDirectionalLight) }]
                }]
        }], propDecorators: { castShadow: [{
                type: Input
            }], position: [{
                type: Input
            }], shadow: [{
                type: Input
            }], target: [{
                type: Input
            }] } });
