// ngx-three/esm2020/lib/generated/ThFlyControls.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { FlyControls } from 'three/examples/jsm/controls/FlyControls';
import { ThControlBase } from '../ThControlBase';
import * as i0 from "@angular/core";
export class ThFlyControls extends ThControlBase {
    getType() {
        return FlyControls;
    }
    set object(value) {
        if (this._objRef) {
            this._objRef.object = value;
        }
    }
    // @ts-ignore
    get object() {
        return this._objRef?.object;
    }
    set domElement(value) {
        if (this._objRef) {
            this._objRef.domElement = value;
        }
    }
    // @ts-ignore
    get domElement() {
        return this._objRef?.domElement;
    }
    set movementSpeed(value) {
        if (this._objRef) {
            this._objRef.movementSpeed = value;
        }
    }
    // @ts-ignore
    get movementSpeed() {
        return this._objRef?.movementSpeed;
    }
    set rollSpeed(value) {
        if (this._objRef) {
            this._objRef.rollSpeed = value;
        }
    }
    // @ts-ignore
    get rollSpeed() {
        return this._objRef?.rollSpeed;
    }
    set dragToLook(value) {
        if (this._objRef) {
            this._objRef.dragToLook = value;
        }
    }
    // @ts-ignore
    get dragToLook() {
        return this._objRef?.dragToLook;
    }
    set autoForward(value) {
        if (this._objRef) {
            this._objRef.autoForward = value;
        }
    }
    // @ts-ignore
    get autoForward() {
        return this._objRef?.autoForward;
    }
}
ThFlyControls.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThFlyControls, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThFlyControls.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThFlyControls, selector: "th-flyControls", inputs: { object: "object", domElement: "domElement", movementSpeed: "movementSpeed", rollSpeed: "rollSpeed", dragToLook: "dragToLook", autoForward: "autoForward" }, providers: [{ provide: ThControlBase, useExisting: forwardRef(() => ThFlyControls) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThFlyControls, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-flyControls',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThControlBase, useExisting: forwardRef(() => ThFlyControls) }]
                }]
        }], propDecorators: { object: [{
                type: Input
            }], domElement: [{
                type: Input
            }], movementSpeed: [{
                type: Input
            }], rollSpeed: [{
                type: Input
            }], dragToLook: [{
                type: Input
            }], autoForward: [{
                type: Input
            }] } });
