// ngx-three/esm2020/lib/generated/ThGlitchPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { GlitchPass } from 'three/examples/jsm/postprocessing/GlitchPass';
import { ThPassBase } from '../ThPassBase';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThGlitchPass extends ThPass {
    getType() {
        return GlitchPass;
    }
    set uniforms(map) {
        if (this._objRef) {
            Object.assign(this._objRef.uniforms, map);
        }
    }
    // @ts-ignore
    get uniforms() {
        // @ts-ignore
        return this._objRef?.uniforms;
    }
    set material(value) {
        if (this._objRef) {
            this._objRef.material = value;
        }
    }
    // @ts-ignore
    get material() {
        return this._objRef?.material;
    }
    set fsQuad(value) {
        if (this._objRef) {
            this._objRef.fsQuad = value;
        }
    }
    // @ts-ignore
    get fsQuad() {
        return this._objRef?.fsQuad;
    }
    set goWild(value) {
        if (this._objRef) {
            this._objRef.goWild = value;
        }
    }
    // @ts-ignore
    get goWild() {
        return this._objRef?.goWild;
    }
    set curF(value) {
        if (this._objRef) {
            this._objRef.curF = value;
        }
    }
    // @ts-ignore
    get curF() {
        return this._objRef?.curF;
    }
    set randX(value) {
        if (this._objRef) {
            this._objRef.randX = value;
        }
    }
    // @ts-ignore
    get randX() {
        return this._objRef?.randX;
    }
}
ThGlitchPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGlitchPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThGlitchPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThGlitchPass, selector: "th-glitchPass", inputs: { uniforms: "uniforms", material: "material", fsQuad: "fsQuad", goWild: "goWild", curF: "curF", randX: "randX" }, providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThGlitchPass) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGlitchPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-glitchPass',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThGlitchPass) }]
                }]
        }], propDecorators: { uniforms: [{
                type: Input
            }], material: [{
                type: Input
            }], fsQuad: [{
                type: Input
            }], goWild: [{
                type: Input
            }], curF: [{
                type: Input
            }], randX: [{
                type: Input
            }] } });
