// ngx-three/esm2020/lib/generated/ThIcosahedronGeometry.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { IcosahedronGeometry } from 'three';
import { ThBufferGeometry } from './ThBufferGeometry';
import { ThPolyhedronGeometry } from './ThPolyhedronGeometry';
import * as i0 from "@angular/core";
export class ThIcosahedronGeometry extends ThPolyhedronGeometry {
    getType() {
        return IcosahedronGeometry;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
}
ThIcosahedronGeometry.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThIcosahedronGeometry, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThIcosahedronGeometry.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThIcosahedronGeometry, selector: "th-icosahedronGeometry", providers: [
        {
            provide: ThBufferGeometry,
            useExisting: forwardRef(() => ThIcosahedronGeometry)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThIcosahedronGeometry, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-icosahedronGeometry',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThBufferGeometry,
                            useExisting: forwardRef(() => ThIcosahedronGeometry)
                        }
                    ]
                }]
        }] });
