// ngx-three/esm2020/lib/generated/ThInstancedMesh.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { InstancedMesh } from 'three';
import { applyValue } from '../util';
import { ThMesh } from './ThMesh';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThInstancedMesh extends ThMesh {
    getType() {
        return InstancedMesh;
    }
    // @ts-ignore
    get isInstancedMesh() {
        return this._objRef?.isInstancedMesh;
    }
    set boundingBox(value) {
        if (this._objRef) {
            this._objRef.boundingBox = applyValue(this._objRef.boundingBox, value);
        }
    }
    // @ts-ignore
    get boundingBox() {
        return this._objRef?.boundingBox;
    }
    set boundingSphere(value) {
        if (this._objRef) {
            this._objRef.boundingSphere = applyValue(this._objRef.boundingSphere, value);
        }
    }
    // @ts-ignore
    get boundingSphere() {
        return this._objRef?.boundingSphere;
    }
    set count(value) {
        if (this._objRef) {
            this._objRef.count = value;
        }
    }
    // @ts-ignore
    get count() {
        return this._objRef?.count;
    }
    set instanceColor(value) {
        if (this._objRef) {
            this._objRef.instanceColor = applyValue(this._objRef.instanceColor, value);
        }
    }
    // @ts-ignore
    get instanceColor() {
        return this._objRef?.instanceColor;
    }
    set instanceMatrix(value) {
        if (this._objRef) {
            this._objRef.instanceMatrix = applyValue(this._objRef.instanceMatrix, value);
        }
    }
    // @ts-ignore
    get instanceMatrix() {
        return this._objRef?.instanceMatrix;
    }
}
ThInstancedMesh.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThInstancedMesh, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThInstancedMesh.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThInstancedMesh, selector: "th-instancedMesh", inputs: { boundingBox: "boundingBox", boundingSphere: "boundingSphere", count: "count", instanceColor: "instanceColor", instanceMatrix: "instanceMatrix" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThInstancedMesh) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThInstancedMesh, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-instancedMesh',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThInstancedMesh) }]
                }]
        }], propDecorators: { boundingBox: [{
                type: Input
            }], boundingSphere: [{
                type: Input
            }], count: [{
                type: Input
            }], instanceColor: [{
                type: Input
            }], instanceMatrix: [{
                type: Input
            }] } });
