// ngx-three/esm2020/lib/generated/ThLineSegments.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { LineSegments } from 'three';
import { ThLine } from './ThLine';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThLineSegments extends ThLine {
    getType() {
        return LineSegments;
    }
    // @ts-ignore
    get isLineSegments() {
        return this._objRef?.isLineSegments;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
}
ThLineSegments.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLineSegments, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLineSegments.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThLineSegments, selector: "th-lineSegments", providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLineSegments) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLineSegments, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lineSegments',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLineSegments) }]
                }]
        }] });
