// ngx-three/esm2020/lib/generated/ThLightningStrike.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { LightningStrike } from 'three/examples/jsm/geometries/LightningStrike';
import { ThBufferGeometry } from './ThBufferGeometry';
import * as i0 from "@angular/core";
export class ThLightningStrike extends ThBufferGeometry {
    getType() {
        return LightningStrike;
    }
    set state(value) {
        if (this._objRef) {
            this._objRef.state = value;
        }
    }
    // @ts-ignore
    get state() {
        return this._objRef?.state;
    }
}
ThLightningStrike.RAY_INITIALIZED = LightningStrike.RAY_INITIALIZED;
ThLightningStrike.RAY_UNBORN = LightningStrike.RAY_UNBORN;
ThLightningStrike.RAY_PROPAGATING = LightningStrike.RAY_PROPAGATING;
ThLightningStrike.RAY_STEADY = LightningStrike.RAY_STEADY;
ThLightningStrike.RAY_VANISHING = LightningStrike.RAY_VANISHING;
ThLightningStrike.RAY_EXTINGUISHED = LightningStrike.RAY_EXTINGUISHED;
ThLightningStrike.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLightningStrike, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLightningStrike.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThLightningStrike, selector: "th-lightningStrike", inputs: { state: "state" }, providers: [
        {
            provide: ThBufferGeometry,
            useExisting: forwardRef(() => ThLightningStrike)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLightningStrike, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lightningStrike',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThBufferGeometry,
                            useExisting: forwardRef(() => ThLightningStrike)
                        }
                    ]
                }]
        }], propDecorators: { state: [{
                type: Input
            }] } });
