// ngx-three/esm2020/lib/generated/ThLineLoop.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { LineLoop } from 'three';
import { ThLine } from './ThLine';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThLineLoop extends ThLine {
    getType() {
        return LineLoop;
    }
    // @ts-ignore
    get isLineLoop() {
        return this._objRef?.isLineLoop;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
}
ThLineLoop.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLineLoop, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLineLoop.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThLineLoop, selector: "th-lineLoop", providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLineLoop) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLineLoop, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lineLoop',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLineLoop) }]
                }]
        }] });
