// ngx-three/esm2020/lib/generated/ThLineDashedMaterial.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { LineDashedMaterial } from 'three';
import { ThLineBasicMaterial } from './ThLineBasicMaterial';
import { ThMaterial } from './ThMaterial';
import * as i0 from "@angular/core";
export class ThLineDashedMaterial extends ThLineBasicMaterial {
    getType() {
        return LineDashedMaterial;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set scale(value) {
        if (this._objRef) {
            this._objRef.scale = value;
        }
    }
    // @ts-ignore
    get scale() {
        return this._objRef?.scale;
    }
    set dashSize(value) {
        if (this._objRef) {
            this._objRef.dashSize = value;
        }
    }
    // @ts-ignore
    get dashSize() {
        return this._objRef?.dashSize;
    }
    set gapSize(value) {
        if (this._objRef) {
            this._objRef.gapSize = value;
        }
    }
    // @ts-ignore
    get gapSize() {
        return this._objRef?.gapSize;
    }
    // @ts-ignore
    get isLineDashedMaterial() {
        return this._objRef?.isLineDashedMaterial;
    }
}
ThLineDashedMaterial.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLineDashedMaterial, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLineDashedMaterial.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThLineDashedMaterial, selector: "th-lineDashedMaterial", inputs: { type: "type", scale: "scale", dashSize: "dashSize", gapSize: "gapSize" }, providers: [
        {
            provide: ThMaterial,
            useExisting: forwardRef(() => ThLineDashedMaterial)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLineDashedMaterial, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lineDashedMaterial',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThMaterial,
                            useExisting: forwardRef(() => ThLineDashedMaterial)
                        }
                    ]
                }]
        }], propDecorators: { type: [{
                type: Input
            }], scale: [{
                type: Input
            }], dashSize: [{
                type: Input
            }], gapSize: [{
                type: Input
            }] } });
