// ngx-three/esm2020/lib/generated/ThLOD.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { LOD } from 'three';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThLOD extends ThObject3D {
    getType() {
        return LOD;
    }
    // @ts-ignore
    get isLOD() {
        return this._objRef?.isLOD;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set levels(value) {
        if (this._objRef) {
            this._objRef.levels = value;
        }
    }
    // @ts-ignore
    get levels() {
        return this._objRef?.levels;
    }
    set autoUpdate(value) {
        if (this._objRef) {
            this._objRef.autoUpdate = value;
        }
    }
    // @ts-ignore
    get autoUpdate() {
        return this._objRef?.autoUpdate;
    }
}
ThLOD.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLOD, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThLOD.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThLOD, selector: "th-lOD", inputs: { levels: "levels", autoUpdate: "autoUpdate" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLOD) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLOD, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-lOD',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThLOD) }]
                }]
        }], propDecorators: { levels: [{
                type: Input
            }], autoUpdate: [{
                type: Input
            }] } });
