// ngx-three/esm2020/lib/generated/ThMapControls.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { MapControls } from 'three/examples/jsm/controls/MapControls';
import { ThControlBase } from '../ThControlBase';
import { ThOrbitControls } from './ThOrbitControls';
import * as i0 from "@angular/core";
export class ThMapControls extends ThOrbitControls {
    getType() {
        return MapControls;
    }
}
ThMapControls.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThMapControls, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThMapControls.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThMapControls, selector: "th-mapControls", providers: [{ provide: ThControlBase, useExisting: forwardRef(() => ThMapControls) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThMapControls, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-mapControls',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThControlBase, useExisting: forwardRef(() => ThMapControls) }]
                }]
        }] });
