// ngx-three/esm2020/lib/generated/ThPlaneHelper.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { PlaneHelper } from 'three';
import { applyValue } from '../util';
import { ThLineSegments } from './ThLineSegments';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThPlaneHelper extends ThLineSegments {
    getType() {
        return PlaneHelper;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set plane(value) {
        if (this._objRef) {
            this._objRef.plane = applyValue(this._objRef.plane, value);
        }
    }
    // @ts-ignore
    get plane() {
        return this._objRef?.plane;
    }
    set size(value) {
        if (this._objRef) {
            this._objRef.size = value;
        }
    }
    // @ts-ignore
    get size() {
        return this._objRef?.size;
    }
}
ThPlaneHelper.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPlaneHelper, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThPlaneHelper.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThPlaneHelper, selector: "th-planeHelper", inputs: { plane: "plane", size: "size" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThPlaneHelper) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPlaneHelper, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-planeHelper',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThPlaneHelper) }]
                }]
        }], propDecorators: { plane: [{
                type: Input
            }], size: [{
                type: Input
            }] } });
