// ngx-three/esm2020/lib/generated/ThPointLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { PointLight } from 'three';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThPointLight extends ThLight {
    getType() {
        return PointLight;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set intensity(value) {
        if (this._objRef) {
            this._objRef.intensity = value;
        }
    }
    // @ts-ignore
    get intensity() {
        return this._objRef?.intensity;
    }
    set distance(value) {
        if (this._objRef) {
            this._objRef.distance = value;
        }
    }
    // @ts-ignore
    get distance() {
        return this._objRef?.distance;
    }
    set castShadow(value) {
        if (this._objRef) {
            this._objRef.castShadow = value;
        }
    }
    // @ts-ignore
    get castShadow() {
        return this._objRef?.castShadow;
    }
    set decay(value) {
        if (this._objRef) {
            this._objRef.decay = value;
        }
    }
    // @ts-ignore
    get decay() {
        return this._objRef?.decay;
    }
    set shadow(value) {
        if (this._objRef) {
            this._objRef.shadow = value;
        }
    }
    // @ts-ignore
    get shadow() {
        return this._objRef?.shadow;
    }
    set power(value) {
        if (this._objRef) {
            this._objRef.power = value;
        }
    }
    // @ts-ignore
    get power() {
        return this._objRef?.power;
    }
}
ThPointLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPointLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThPointLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThPointLight, selector: "th-pointLight", inputs: { type: "type", intensity: "intensity", distance: "distance", castShadow: "castShadow", decay: "decay", shadow: "shadow", power: "power" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThPointLight) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPointLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-pointLight',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThPointLight) }]
                }]
        }], propDecorators: { type: [{
                type: Input
            }], intensity: [{
                type: Input
            }], distance: [{
                type: Input
            }], castShadow: [{
                type: Input
            }], decay: [{
                type: Input
            }], shadow: [{
                type: Input
            }], power: [{
                type: Input
            }] } });
