// ngx-three/esm2020/lib/generated/ThRenderPassGen.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { RenderPass } from 'three/examples/jsm/postprocessing/RenderPass';
import { ThPassBase } from '../ThPassBase';
import { applyValue } from '../util';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThRenderPassGen extends ThPass {
    getType() {
        return RenderPass;
    }
    set scene(value) {
        if (this._objRef) {
            this._objRef.scene = value;
        }
    }
    // @ts-ignore
    get scene() {
        return this._objRef?.scene;
    }
    set camera(value) {
        if (this._objRef) {
            this._objRef.camera = value;
        }
    }
    // @ts-ignore
    get camera() {
        return this._objRef?.camera;
    }
    set overrideMaterial(value) {
        if (this._objRef) {
            this._objRef.overrideMaterial = value;
        }
    }
    // @ts-ignore
    get overrideMaterial() {
        return this._objRef?.overrideMaterial;
    }
    set clearColor(value) {
        if (this._objRef) {
            this._objRef.clearColor = applyValue(this._objRef.clearColor, value);
        }
    }
    // @ts-ignore
    get clearColor() {
        return this._objRef?.clearColor;
    }
    set clearAlpha(value) {
        if (this._objRef) {
            this._objRef.clearAlpha = value;
        }
    }
    // @ts-ignore
    get clearAlpha() {
        return this._objRef?.clearAlpha;
    }
    set clearDepth(value) {
        if (this._objRef) {
            this._objRef.clearDepth = value;
        }
    }
    // @ts-ignore
    get clearDepth() {
        return this._objRef?.clearDepth;
    }
}
ThRenderPassGen.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRenderPassGen, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThRenderPassGen.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThRenderPassGen, selector: "th-renderPassGen", inputs: { scene: "scene", camera: "camera", overrideMaterial: "overrideMaterial", clearColor: "clearColor", clearAlpha: "clearAlpha", clearDepth: "clearDepth" }, providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThRenderPassGen) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRenderPassGen, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-renderPassGen',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThRenderPassGen) }]
                }]
        }], propDecorators: { scene: [{
                type: Input
            }], camera: [{
                type: Input
            }], overrideMaterial: [{
                type: Input
            }], clearColor: [{
                type: Input
            }], clearAlpha: [{
                type: Input
            }], clearDepth: [{
                type: Input
            }] } });
