// ngx-three/esm2020/lib/generated/ThRectAreaLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { RectAreaLight } from 'three';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThRectAreaLight extends ThLight {
    getType() {
        return RectAreaLight;
    }
    // @ts-ignore
    get isRectAreaLight() {
        return this._objRef?.isRectAreaLight;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set width(value) {
        if (this._objRef) {
            this._objRef.width = value;
        }
    }
    // @ts-ignore
    get width() {
        return this._objRef?.width;
    }
    set height(value) {
        if (this._objRef) {
            this._objRef.height = value;
        }
    }
    // @ts-ignore
    get height() {
        return this._objRef?.height;
    }
    set intensity(value) {
        if (this._objRef) {
            this._objRef.intensity = value;
        }
    }
    // @ts-ignore
    get intensity() {
        return this._objRef?.intensity;
    }
    set power(value) {
        if (this._objRef) {
            this._objRef.power = value;
        }
    }
    // @ts-ignore
    get power() {
        return this._objRef?.power;
    }
}
ThRectAreaLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRectAreaLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThRectAreaLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThRectAreaLight, selector: "th-rectAreaLight", inputs: { width: "width", height: "height", intensity: "intensity", power: "power" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThRectAreaLight) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRectAreaLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-rectAreaLight',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThRectAreaLight) }]
                }]
        }], propDecorators: { width: [{
                type: Input
            }], height: [{
                type: Input
            }], intensity: [{
                type: Input
            }], power: [{
                type: Input
            }] } });
