// ngx-three/esm2020/lib/generated/ThSAOPass.mjs
/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { SAOPass } from 'three/examples/jsm/postprocessing/SAOPass';
import { ThPassBase } from '../ThPassBase';
import { applyValue } from '../util';
import { ThPass } from './ThPass';
import * as i0 from "@angular/core";
export class ThSAOPass extends ThPass {
    getType() {
        return SAOPass;
    }
    set scene(value) {
        if (this._objRef) {
            this._objRef.scene = value;
        }
    }
    // @ts-ignore
    get scene() {
        return this._objRef?.scene;
    }
    set camera(value) {
        if (this._objRef) {
            this._objRef.camera = value;
        }
    }
    // @ts-ignore
    get camera() {
        return this._objRef?.camera;
    }
    set supportsDepthTextureExtension(value) {
        if (this._objRef) {
            this._objRef.supportsDepthTextureExtension = value;
        }
    }
    // @ts-ignore
    get supportsDepthTextureExtension() {
        return this._objRef?.supportsDepthTextureExtension;
    }
    set supportsNormalTexture(value) {
        if (this._objRef) {
            this._objRef.supportsNormalTexture = value;
        }
    }
    // @ts-ignore
    get supportsNormalTexture() {
        return this._objRef?.supportsNormalTexture;
    }
    set originalClearColor(value) {
        if (this._objRef) {
            this._objRef.originalClearColor = applyValue(this._objRef.originalClearColor, value);
        }
    }
    // @ts-ignore
    get originalClearColor() {
        return this._objRef?.originalClearColor;
    }
    set oldClearColor(value) {
        if (this._objRef) {
            this._objRef.oldClearColor = applyValue(this._objRef.oldClearColor, value);
        }
    }
    // @ts-ignore
    get oldClearColor() {
        return this._objRef?.oldClearColor;
    }
    set oldClearAlpha(value) {
        if (this._objRef) {
            this._objRef.oldClearAlpha = value;
        }
    }
    // @ts-ignore
    get oldClearAlpha() {
        return this._objRef?.oldClearAlpha;
    }
    set resolution(value) {
        if (this._objRef) {
            this._objRef.resolution = applyValue(this._objRef.resolution, value);
        }
    }
    // @ts-ignore
    get resolution() {
        return this._objRef?.resolution;
    }
    set saoRenderTarget(value) {
        if (this._objRef) {
            this._objRef.saoRenderTarget = value;
        }
    }
    // @ts-ignore
    get saoRenderTarget() {
        return this._objRef?.saoRenderTarget;
    }
    set blurIntermediateRenderTarget(value) {
        if (this._objRef) {
            this._objRef.blurIntermediateRenderTarget = value;
        }
    }
    // @ts-ignore
    get blurIntermediateRenderTarget() {
        return this._objRef?.blurIntermediateRenderTarget;
    }
    set beautyRenderTarget(value) {
        if (this._objRef) {
            this._objRef.beautyRenderTarget = value;
        }
    }
    // @ts-ignore
    get beautyRenderTarget() {
        return this._objRef?.beautyRenderTarget;
    }
    set normalRenderTarget(value) {
        if (this._objRef) {
            this._objRef.normalRenderTarget = value;
        }
    }
    // @ts-ignore
    get normalRenderTarget() {
        return this._objRef?.normalRenderTarget;
    }
    set depthRenderTarget(value) {
        if (this._objRef) {
            this._objRef.depthRenderTarget = value;
        }
    }
    // @ts-ignore
    get depthRenderTarget() {
        return this._objRef?.depthRenderTarget;
    }
    set depthMaterial(value) {
        if (this._objRef) {
            this._objRef.depthMaterial = value;
        }
    }
    // @ts-ignore
    get depthMaterial() {
        return this._objRef?.depthMaterial;
    }
    set normalMaterial(value) {
        if (this._objRef) {
            this._objRef.normalMaterial = value;
        }
    }
    // @ts-ignore
    get normalMaterial() {
        return this._objRef?.normalMaterial;
    }
    set saoMaterial(value) {
        if (this._objRef) {
            this._objRef.saoMaterial = value;
        }
    }
    // @ts-ignore
    get saoMaterial() {
        return this._objRef?.saoMaterial;
    }
    set vBlurMaterial(value) {
        if (this._objRef) {
            this._objRef.vBlurMaterial = value;
        }
    }
    // @ts-ignore
    get vBlurMaterial() {
        return this._objRef?.vBlurMaterial;
    }
    set hBlurMaterial(value) {
        if (this._objRef) {
            this._objRef.hBlurMaterial = value;
        }
    }
    // @ts-ignore
    get hBlurMaterial() {
        return this._objRef?.hBlurMaterial;
    }
    set materialCopy(value) {
        if (this._objRef) {
            this._objRef.materialCopy = value;
        }
    }
    // @ts-ignore
    get materialCopy() {
        return this._objRef?.materialCopy;
    }
    set depthCopy(value) {
        if (this._objRef) {
            this._objRef.depthCopy = value;
        }
    }
    // @ts-ignore
    get depthCopy() {
        return this._objRef?.depthCopy;
    }
    set fsQuad(value) {
        if (this._objRef) {
            this._objRef.fsQuad = value;
        }
    }
    // @ts-ignore
    get fsQuad() {
        return this._objRef?.fsQuad;
    }
    set params(value) {
        if (this._objRef) {
            this._objRef.params = value;
        }
    }
    // @ts-ignore
    get params() {
        return this._objRef?.params;
    }
}
ThSAOPass.OUTPUT = SAOPass.OUTPUT;
ThSAOPass.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThSAOPass, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThSAOPass.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThSAOPass, selector: "th-sAOPass", inputs: { scene: "scene", camera: "camera", supportsDepthTextureExtension: "supportsDepthTextureExtension", supportsNormalTexture: "supportsNormalTexture", originalClearColor: "originalClearColor", oldClearColor: "oldClearColor", oldClearAlpha: "oldClearAlpha", resolution: "resolution", saoRenderTarget: "saoRenderTarget", blurIntermediateRenderTarget: "blurIntermediateRenderTarget", beautyRenderTarget: "beautyRenderTarget", normalRenderTarget: "normalRenderTarget", depthRenderTarget: "depthRenderTarget", depthMaterial: "depthMaterial", normalMaterial: "normalMaterial", saoMaterial: "saoMaterial", vBlurMaterial: "vBlurMaterial", hBlurMaterial: "hBlurMaterial", materialCopy: "materialCopy", depthCopy: "depthCopy", fsQuad: "fsQuad", params: "params" }, providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThSAOPass) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThSAOPass, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-sAOPass',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThPassBase, useExisting: forwardRef(() => ThSAOPass) }]
                }]
        }], propDecorators: { scene: [{
                type: Input
            }], camera: [{
                type: Input
            }], supportsDepthTextureExtension: [{
                type: Input
            }], supportsNormalTexture: [{
                type: Input
            }], originalClearColor: [{
                type: Input
            }], oldClearColor: [{
                type: Input
            }], oldClearAlpha: [{
                type: Input
            }], resolution: [{
                type: Input
            }], saoRenderTarget: [{
                type: Input
            }], blurIntermediateRenderTarget: [{
                type: Input
            }], beautyRenderTarget: [{
                type: Input
            }], normalRenderTarget: [{
                type: Input
            }], depthRenderTarget: [{
                type: Input
            }], depthMaterial: [{
                type: Input
            }], normalMaterial: [{
                type: Input
            }], saoMaterial: [{
                type: Input
            }], vBlurMaterial: [{
                type: Input
            }], hBlurMaterial: [{
                type: Input
            }], materialCopy: [{
                type: Input
            }], depthCopy: [{
                type: Input
            }], fsQuad: [{
                type: Input
            }], params: [{
                type: Input
            }] } });
