// ngx-three/esm2020/lib/generated/ThScene.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { Scene } from 'three';
import { applyValue } from '../util';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThScene extends ThObject3D {
    getType() {
        return Scene;
    }
    // @ts-ignore
    get isScene() {
        return this._objRef?.isScene;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set fog(value) {
        if (this._objRef) {
            this._objRef.fog = value;
        }
    }
    // @ts-ignore
    get fog() {
        return this._objRef?.fog;
    }
    set backgroundBlurriness(value) {
        if (this._objRef) {
            this._objRef.backgroundBlurriness = value;
        }
    }
    // @ts-ignore
    get backgroundBlurriness() {
        return this._objRef?.backgroundBlurriness;
    }
    set backgroundIntensity(value) {
        if (this._objRef) {
            this._objRef.backgroundIntensity = value;
        }
    }
    // @ts-ignore
    get backgroundIntensity() {
        return this._objRef?.backgroundIntensity;
    }
    set overrideMaterial(value) {
        if (this._objRef) {
            this._objRef.overrideMaterial = value;
        }
    }
    // @ts-ignore
    get overrideMaterial() {
        return this._objRef?.overrideMaterial;
    }
    set background(value) {
        if (this._objRef) {
            this._objRef.background = applyValue(this._objRef.background, value);
        }
    }
    // @ts-ignore
    get background() {
        return this._objRef?.background;
    }
    set environment(value) {
        if (this._objRef) {
            this._objRef.environment = value;
        }
    }
    // @ts-ignore
    get environment() {
        return this._objRef?.environment;
    }
}
ThScene.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThScene, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThScene.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThScene, selector: "th-scene", inputs: { type: "type", fog: "fog", backgroundBlurriness: "backgroundBlurriness", backgroundIntensity: "backgroundIntensity", overrideMaterial: "overrideMaterial", background: "background", environment: "environment" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThScene) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThScene, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-scene',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThScene) }]
                }]
        }], propDecorators: { type: [{
                type: Input
            }], fog: [{
                type: Input
            }], backgroundBlurriness: [{
                type: Input
            }], backgroundIntensity: [{
                type: Input
            }], overrideMaterial: [{
                type: Input
            }], background: [{
                type: Input
            }], environment: [{
                type: Input
            }] } });
