// ngx-three/esm2020/lib/generated/ThSpotLight.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { SpotLight } from 'three';
import { applyValue } from '../util';
import { ThLight } from './ThLight';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThSpotLight extends ThLight {
    getType() {
        return SpotLight;
    }
    // @ts-ignore
    get isSpotLight() {
        return this._objRef?.isSpotLight;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set position(value) {
        if (this._objRef) {
            this._objRef.position = applyValue(this._objRef.position, value);
        }
    }
    // @ts-ignore
    get position() {
        return this._objRef?.position;
    }
    set target(value) {
        if (this._objRef) {
            this._objRef.target = value;
        }
    }
    // @ts-ignore
    get target() {
        return this._objRef?.target;
    }
    set castShadow(value) {
        if (this._objRef) {
            this._objRef.castShadow = value;
        }
    }
    // @ts-ignore
    get castShadow() {
        return this._objRef?.castShadow;
    }
    set intensity(value) {
        if (this._objRef) {
            this._objRef.intensity = value;
        }
    }
    // @ts-ignore
    get intensity() {
        return this._objRef?.intensity;
    }
    set distance(value) {
        if (this._objRef) {
            this._objRef.distance = value;
        }
    }
    // @ts-ignore
    get distance() {
        return this._objRef?.distance;
    }
    set angle(value) {
        if (this._objRef) {
            this._objRef.angle = value;
        }
    }
    // @ts-ignore
    get angle() {
        return this._objRef?.angle;
    }
    set decay(value) {
        if (this._objRef) {
            this._objRef.decay = value;
        }
    }
    // @ts-ignore
    get decay() {
        return this._objRef?.decay;
    }
    set shadow(value) {
        if (this._objRef) {
            this._objRef.shadow = value;
        }
    }
    // @ts-ignore
    get shadow() {
        return this._objRef?.shadow;
    }
    set power(value) {
        if (this._objRef) {
            this._objRef.power = value;
        }
    }
    // @ts-ignore
    get power() {
        return this._objRef?.power;
    }
    set penumbra(value) {
        if (this._objRef) {
            this._objRef.penumbra = value;
        }
    }
    // @ts-ignore
    get penumbra() {
        return this._objRef?.penumbra;
    }
    set map(value) {
        if (this._objRef) {
            this._objRef.map = value;
        }
    }
    // @ts-ignore
    get map() {
        return this._objRef?.map;
    }
}
ThSpotLight.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThSpotLight, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThSpotLight.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThSpotLight, selector: "th-spotLight", inputs: { position: "position", target: "target", castShadow: "castShadow", intensity: "intensity", distance: "distance", angle: "angle", decay: "decay", shadow: "shadow", power: "power", penumbra: "penumbra", map: "map" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThSpotLight) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThSpotLight, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-spotLight',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThSpotLight) }]
                }]
        }], propDecorators: { position: [{
                type: Input
            }], target: [{
                type: Input
            }], castShadow: [{
                type: Input
            }], intensity: [{
                type: Input
            }], distance: [{
                type: Input
            }], angle: [{
                type: Input
            }], decay: [{
                type: Input
            }], shadow: [{
                type: Input
            }], power: [{
                type: Input
            }], penumbra: [{
                type: Input
            }], map: [{
                type: Input
            }] } });
