// ngx-three/esm2020/lib/generated/ThSprite.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { Sprite } from 'three';
import { applyValue } from '../util';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThSprite extends ThObject3D {
    getType() {
        return Sprite;
    }
    // @ts-ignore
    get isSprite() {
        return this._objRef?.isSprite;
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set castShadow(value) {
        if (this._objRef) {
            this._objRef.castShadow = value;
        }
    }
    // @ts-ignore
    get castShadow() {
        return this._objRef?.castShadow;
    }
    set geometry(value) {
        if (this._objRef) {
            this._objRef.geometry = value;
        }
    }
    // @ts-ignore
    get geometry() {
        return this._objRef?.geometry;
    }
    set material(value) {
        if (this._objRef) {
            this._objRef.material = value;
        }
    }
    // @ts-ignore
    get material() {
        return this._objRef?.material;
    }
    set center(value) {
        if (this._objRef) {
            this._objRef.center = applyValue(this._objRef.center, value);
        }
    }
    // @ts-ignore
    get center() {
        return this._objRef?.center;
    }
}
ThSprite.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThSprite, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThSprite.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThSprite, selector: "th-sprite", inputs: { castShadow: "castShadow", geometry: "geometry", material: "material", center: "center" }, providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThSprite) }], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThSprite, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-sprite',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [{ provide: ThObject3D, useExisting: forwardRef(() => ThSprite) }]
                }]
        }], propDecorators: { castShadow: [{
                type: Input
            }], geometry: [{
                type: Input
            }], material: [{
                type: Input
            }], center: [{
                type: Input
            }] } });
