// ngx-three/esm2020/lib/generated/ThStereoCamera.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { StereoCamera } from 'three';
import { ThCamera } from './ThCamera';
import { ThObject3D } from './ThObject3D';
import * as i0 from "@angular/core";
export class ThStereoCamera extends ThCamera {
    getType() {
        return StereoCamera;
    }
    set type(value) {
        if (this._objRef) {
            this._objRef.type = value;
        }
    }
    // @ts-ignore
    get type() {
        return this._objRef?.type;
    }
    set aspect(value) {
        if (this._objRef) {
            this._objRef.aspect = value;
        }
    }
    // @ts-ignore
    get aspect() {
        return this._objRef?.aspect;
    }
    set eyeSep(value) {
        if (this._objRef) {
            this._objRef.eyeSep = value;
        }
    }
    // @ts-ignore
    get eyeSep() {
        return this._objRef?.eyeSep;
    }
    set cameraL(value) {
        if (this._objRef) {
            this._objRef.cameraL = value;
        }
    }
    // @ts-ignore
    get cameraL() {
        return this._objRef?.cameraL;
    }
    set cameraR(value) {
        if (this._objRef) {
            this._objRef.cameraR = value;
        }
    }
    // @ts-ignore
    get cameraR() {
        return this._objRef?.cameraR;
    }
}
ThStereoCamera.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThStereoCamera, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThStereoCamera.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThStereoCamera, selector: "th-stereoCamera", inputs: { type: "type", aspect: "aspect", eyeSep: "eyeSep", cameraL: "cameraL", cameraR: "cameraR" }, providers: [
        { provide: ThObject3D, useExisting: forwardRef(() => ThStereoCamera) },
        { provide: ThCamera, useExisting: forwardRef(() => ThStereoCamera) }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThStereoCamera, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-stereoCamera',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        { provide: ThObject3D, useExisting: forwardRef(() => ThStereoCamera) },
                        { provide: ThCamera, useExisting: forwardRef(() => ThStereoCamera) }
                    ]
                }]
        }], propDecorators: { type: [{
                type: Input
            }], aspect: [{
                type: Input
            }], eyeSep: [{
                type: Input
            }], cameraL: [{
                type: Input
            }], cameraR: [{
                type: Input
            }] } });
