// ngx-three/esm2020/lib/events/raycaster.events.directive.mjs
import { Directive, EventEmitter, Host, Output } from '@angular/core';
import { RaycasterEvent } from './raycaster.service';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThObject3D";
import * as i2 from "./raycaster.service";
export class RaycasterEventDirective {
    get onMouseEnter() {
        if (!this.mouseEnter) {
            this.mouseEnter = new EventEmitter();
        }
        return this.mouseEnter;
    }
    get onMouseExit() {
        if (!this.mouseExit) {
            this.mouseExit = new EventEmitter();
        }
        return this.mouseExit;
    }
    get onClick() {
        if (!this.click) {
            this.click = new EventEmitter();
        }
        return this.click;
    }
    get onPointerDown() {
        if (!this.pointerDown) {
            this.pointerDown = new EventEmitter();
        }
        return this.pointerDown;
    }
    get onPointerUp() {
        if (!this.pointerUp) {
            this.pointerUp = new EventEmitter();
        }
        return this.pointerUp;
    }
    constructor(host, raycasterService) {
        this.host = host;
        this.raycasterService = raycasterService;
        this.emitOnMouseEnter = this.emitOnMouseEnter.bind(this);
        this.emitOnMouseExit = this.emitOnMouseExit.bind(this);
        this.emitOnClick = this.emitOnClick.bind(this);
    }
    ngAfterViewInit() {
        this.subscribeEvents();
    }
    subscribeEvents() {
        this.raycasterService.addEventTarget(this);
    }
    unSubscribeEvents() {
        this.raycasterService.removeEventTarget(this);
    }
    emitOnMouseExit() {
        this.mouseExit?.emit({
            component: this.host,
            type: RaycasterEvent.mouseExit
        });
    }
    emitOnMouseEnter(event) {
        // console.log('RaycasterGroupDirective.onMouseEnter', event);
        this.mouseEnter?.emit(event);
    }
    emitOnClick(event) {
        // console.log('onClick', event);
        this.click?.emit(event);
    }
    emitOnPointerDown(event) {
        // console.log('onClick', event);
        this.pointerDown?.emit(event);
    }
    emitOnPointerUp(event) {
        // console.log('onClick', event);
        this.pointerUp?.emit(event);
    }
    ngOnDestroy() {
        this.unSubscribeEvents();
    }
}
RaycasterEventDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RaycasterEventDirective, deps: [{ token: i1.ThObject3D, host: true }, { token: i2.RaycasterService }], target: i0.ɵɵFactoryTarget.Directive });
RaycasterEventDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: RaycasterEventDirective, selector: "[onClick], [onMouseEnter], [onMouseExit], [onPointerDown], [onPointerUp]", outputs: { onMouseEnter: "onMouseEnter", onMouseExit: "onMouseExit", onClick: "onClick", onPointerDown: "onPointerDown", onPointerUp: "onPointerUp" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RaycasterEventDirective, decorators: [{
            type: Directive,
            args: [{ selector: '[onClick], [onMouseEnter], [onMouseExit], [onPointerDown], [onPointerUp]' }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D, decorators: [{
                    type: Host
                }] }, { type: i2.RaycasterService }]; }, propDecorators: { onMouseEnter: [{
                type: Output
            }], onMouseExit: [{
                type: Output
            }], onClick: [{
                type: Output
            }], onPointerDown: [{
                type: Output
            }], onPointerUp: [{
                type: Output
            }] } });
