// ngx-three/esm2020/lib/loaders/compressed-texture/ThDDSLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { DDSLoader } from 'three/examples/jsm/loaders/DDSLoader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThCompressedTexture";
export class DDSLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = DDSLoader;
    }
}
DDSLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: DDSLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
DDSLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: DDSLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: DDSLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThDDSLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThDDSLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDDSLoaderPipe, deps: [{ token: DDSLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThDDSLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThDDSLoaderPipe, name: "loadDDSTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDDSLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadDDSTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: DDSLoaderService }]; } });
export class ThDDSLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThDDSLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDDSLoaderDirective, deps: [{ token: i1.ThCompressedTexture, host: true }, { token: i0.NgZone }, { token: DDSLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThDDSLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThDDSLoaderDirective, selector: "[loadDDSTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThDDSLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadDDSTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThCompressedTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: DDSLoaderService }]; } });
