// ngx-three/esm2020/lib/loaders/compressed-texture/ThDRACOLoader.mjs
import { Injectable } from '@angular/core';
import { DRACOLoader } from 'three/examples/jsm/loaders/DRACOLoader';
import { ThCallbackLoaderService } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
export class DRACOLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = DRACOLoader;
        this.decoderPath = '';
        this.reuseInstance = true;
    }
    setDecoderPath(path) {
        this.decoderPath = path;
    }
    setDecoderConfig(config) {
        this.decoderConfig = config;
    }
    setWorkerLimit(limit) {
        this.workerLimit = limit;
    }
    setReuseInstance(reuse) {
        this.reuseInstance = reuse;
    }
    createLoaderInstance() {
        if (this.reuseInstance && this.instance) {
            return this.instance;
        }
        const loader = super.createLoaderInstance();
        loader.setDecoderPath(this.decoderPath);
        if (this.decoderConfig) {
            loader.setDecoderConfig(this.decoderConfig);
        }
        if (this.workerLimit) {
            loader.setWorkerLimit(this.workerLimit);
        }
        if (this.reuseInstance) {
            this.instance = loader;
        }
        return loader;
    }
}
DRACOLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: DRACOLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
DRACOLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: DRACOLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: DRACOLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
