// ngx-three/esm2020/lib/loaders/compressed-texture/ThKTX2Loader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { KTX2Loader } from 'three/examples/jsm/loaders/KTX2Loader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThTexture";
export class KTX2LoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = KTX2Loader;
        this.transcoderPath = '';
    }
    setDecoderPath(path) {
        this.transcoderPath = path;
    }
    createLoaderInstance() {
        const loader = super.createLoaderInstance();
        loader.setTranscoderPath(this.transcoderPath);
        return loader;
    }
}
KTX2LoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: KTX2LoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
KTX2LoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: KTX2LoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: KTX2LoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThKTX2LoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThKTX2LoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTX2LoaderPipe, deps: [{ token: KTX2LoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThKTX2LoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThKTX2LoaderPipe, name: "loadKTX2Texture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTX2LoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadKTX2Texture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: KTX2LoaderService }]; } });
export class ThKTX2LoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThKTX2LoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTX2LoaderDirective, deps: [{ token: i1.ThTexture, host: true }, { token: i0.NgZone }, { token: KTX2LoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThKTX2LoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThKTX2LoaderDirective, selector: "[loadKTX2Texture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTX2LoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadKTX2Texture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: KTX2LoaderService }]; } });
