// ngx-three/esm2020/lib/loaders/compressed-texture/ThKTXLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { KTXLoader } from 'three/examples/jsm/loaders/KTXLoader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThCompressedTexture";
export class KTXLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = KTXLoader;
    }
}
KTXLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: KTXLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
KTXLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: KTXLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: KTXLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThKTXLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThKTXLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTXLoaderPipe, deps: [{ token: KTXLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThKTXLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThKTXLoaderPipe, name: "loadKTXTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTXLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadKTXTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: KTXLoaderService }]; } });
export class ThKTXLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThKTXLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTXLoaderDirective, deps: [{ token: i1.ThCompressedTexture, host: true }, { token: i0.NgZone }, { token: KTXLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThKTXLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThKTXLoaderDirective, selector: "[loadKTXTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThKTXLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadKTXTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThCompressedTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: KTXLoaderService }]; } });
