// ngx-three/esm2020/lib/loaders/compressed-texture/ThPVRLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { PVRLoader } from 'three/examples/jsm/loaders/PVRLoader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThCompressedTexture";
export class PVRLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = PVRLoader;
    }
}
PVRLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: PVRLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
PVRLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: PVRLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: PVRLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThPVRLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThPVRLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPVRLoaderPipe, deps: [{ token: PVRLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThPVRLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThPVRLoaderPipe, name: "loadPVRTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPVRLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadPVRTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: PVRLoaderService }]; } });
export class ThPVRLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThPVRLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPVRLoaderDirective, deps: [{ token: i1.ThCompressedTexture, host: true }, { token: i0.NgZone }, { token: PVRLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThPVRLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThPVRLoaderDirective, selector: "[loadPVRTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPVRLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadPVRTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThCompressedTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: PVRLoaderService }]; } });
