// ngx-three/esm2020/lib/generated/overrides/ThTransformControls.mjs
/* eslint-disable @typescript-eslint/naming-convention */
/* eslint-disable no-underscore-dangle */
/* eslint-disable @angular-eslint/component-selector, @angular-eslint/component-class-suffix, jsdoc/no-types, import/no-deprecated */
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { TransformControls } from 'three/examples/jsm/controls/TransformControls';
import { ThControlBase } from '../../ThControlBase';
import { ThTransformControlsGen } from '../ThTransformControlsGen';
import * as i0 from "@angular/core";
export class ThTransformControls extends ThTransformControlsGen {
    getType() {
        return TransformControls;
    }
    set object(value) {
        if (this._objRef) {
            this._objRef.object = value;
            if (value) {
                this._objRef.attach(value);
            }
            else {
                this._objRef.detach();
            }
        }
    }
    addToParent() {
        if (this._objRef) {
            this.canvas?.scene?.objRef?.add(this._objRef);
        }
    }
    removeFromParent() {
        if (this._objRef) {
            this.canvas?.scene?.objRef?.remove(this._objRef);
        }
    }
}
ThTransformControls.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTransformControls, deps: null, target: i0.ɵɵFactoryTarget.Component });
ThTransformControls.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThTransformControls, selector: "th-transformControls", inputs: { object: "object" }, providers: [
        {
            provide: ThControlBase,
            useExisting: forwardRef(() => ThTransformControls)
        }
    ], usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTransformControls, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-transformControls',
                    template: '<ng-content/>',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    providers: [
                        {
                            provide: ThControlBase,
                            useExisting: forwardRef(() => ThTransformControls)
                        }
                    ]
                }]
        }], propDecorators: { object: [{
                type: Input
            }] } });
