// ngx-three/esm2020/lib/ThControlBase.mjs
import { Component, inject } from '@angular/core';
import { ThAnimationLoopService } from './renderer/th-animation-loop.service';
import { ThWrapperBase } from './ThWrapperBase';
import * as i0 from "@angular/core";
import * as i1 from "./generated/ThObject3D";
import * as i2 from "./ThCanvas";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThControlBase extends ThWrapperBase {
    constructor(_camera, canvas) {
        super();
        this._camera = _camera;
        this.canvas = canvas;
        this.renderLoop = inject(ThAnimationLoopService);
    }
    createThreeInstance(args) {
        if (!args) {
            args = [this._camera.objRef, this.canvas?.rendererCanvas?.nativeElement];
        }
        const instance = super.createThreeInstance(args);
        this.patchDispatchEvent(instance);
        this.subscribeControlUpdater(instance);
        return instance;
    }
    patchDispatchEvent(dispatcher) {
        if (dispatcher.dispatchEvent) {
            const origMethod = (this.origDispatchEventMethod = dispatcher.dispatchEvent);
            dispatcher.dispatchEvent = (event) => {
                origMethod.call(dispatcher, event);
                // request an animation frame after an event was emitted;
                this.renderLoop.requestAnimationFrame();
            };
        }
    }
    ngOnDestroy() {
        super.ngOnDestroy();
        this.unpatchDispatchEvent();
        this.unsubscribeControlUpdater();
    }
    unpatchDispatchEvent() {
        if (this.origDispatchEventMethod && this._objRef) {
            this._objRef.dispatchEvent = this.origDispatchEventMethod;
        }
    }
    subscribeControlUpdater(control) {
        if (control.update !== undefined) {
            this.beforeRenderSubscription = this.renderLoop.beforeRender$.subscribe((state) => {
                control.update(state.delta);
            });
        }
    }
    unsubscribeControlUpdater() {
        if (this.beforeRenderSubscription) {
            this.beforeRenderSubscription.unsubscribe();
        }
    }
}
ThControlBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThControlBase, deps: [{ token: i1.ThObject3D }, { token: i2.ThCanvas }], target: i0.ɵɵFactoryTarget.Component });
ThControlBase.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThControlBase, selector: "th-abs-control", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThControlBase, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-abs-control',
                    template: ''
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D }, { type: i2.ThCanvas }]; } });
