// ngx-three/esm2020/lib/ThGeometryBase.mjs
import { Component } from '@angular/core';
import { ThWrapperBase } from './ThWrapperBase';
import * as i0 from "@angular/core";
import * as i1 from "./generated/ThObject3D";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThGeometryBase extends ThWrapperBase {
    constructor(parent) {
        super();
        this.parent = parent;
    }
    addToParent() {
        if (!this.parent.objRef) {
            return;
            //throw new Error("parent object doesn't hold a three js object instance");
        }
        this.parent.objRef.geometry = this.objRef;
        if (this.parent.objRef.updateMorphTargets) {
            this.parent.objRef.updateMorphTargets();
        }
    }
    removeFromParent() {
        if (this._objRef && this.parent && this.parent.objRef) {
            this.parent.objRef.geometry = undefined;
        }
    }
}
ThGeometryBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGeometryBase, deps: [{ token: i1.ThObject3D }], target: i0.ɵɵFactoryTarget.Component });
ThGeometryBase.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThGeometryBase, selector: "th-abs-geometry", usesInheritance: true, ngImport: i0, template: '', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGeometryBase, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-abs-geometry',
                    template: ''
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D }]; } });
