// ngx-three/esm2020/lib/ThObjectBase.mjs
import { Component, Input } from '@angular/core';
import { ThWrapperBase } from './ThWrapperBase';
import * as i0 from "@angular/core";
// eslint-disable-next-line @angular-eslint/component-class-suffix
export class ThObjectBase extends ThWrapperBase {
    constructor(parent) {
        super();
        this.parent = parent;
    }
    addToParent() {
        if (this._objRef &&
            this.parent.objRef &&
            (!this._objRef.parent || this._objRef.parent.uuid !== this.parent.objRef.uuid)) {
            this.parent.objRef.add(this._objRef);
        }
    }
    removeFromParent() {
        this._objRef?.parent?.remove(this._objRef);
    }
    applyObjRef(objRef) {
        if (this._objRef !== objRef || this._objRef?.parent?.uuid !== this.parent.objRef.uuid) {
            this.removeFromParent();
            this._objRef = objRef;
            if (this.autoAddToParent) {
                this.addToParent();
            }
        }
        this.emitObjRefChange();
    }
    attachToParent(newRef, oldRef) {
        if (!this.parent.objRef || (newRef === oldRef && oldRef?.parent?.uuid === this.parent.objRef.uuid)) {
            return;
        }
        // remove old obj from parent
        if (oldRef && oldRef.parent) {
            oldRef.parent.remove(oldRef);
        }
        // add new obj to parent
        if (newRef && (!newRef.parent || (newRef.parent && newRef.parent.uuid !== this.parent.objRef.uuid))) {
            this.parent.objRef.add(newRef);
        }
    }
    // object 3d methods
    set lookAt(vector) {
        Promise.resolve().then(() => {
            // execute next microtick, to assume all
            // position changes already happend,
            // because lookAt triggers a world matrix calculation
            if (!this.objRef) {
                return;
            }
            if (Array.isArray(vector)) {
                this.objRef.lookAt(...vector);
            }
            else {
                this.objRef.lookAt(vector);
            }
        });
    }
}
ThObjectBase.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThObjectBase, deps: [{ token: ThObjectBase }], target: i0.ɵɵFactoryTarget.Component });
ThObjectBase.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "15.1.1", type: ThObjectBase, selector: "th-abs-object", inputs: { lookAt: "lookAt" }, usesInheritance: true, ngImport: i0, template: '<ng-content/>', isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThObjectBase, decorators: [{
            type: Component,
            args: [{
                    selector: 'th-abs-object',
                    template: '<ng-content/>'
                }]
        }], ctorParameters: function () { return [{ type: ThObjectBase }]; }, propDecorators: { lookAt: [{
                type: Input
            }] } });
