// ngx-three/esm2020/lib/loaders/ThCallbackLoaderBase.mjs
import { Directive, EventEmitter, Host, Input, Output } from '@angular/core';
import { isObserved } from '../util';
import { ThLoader } from './ThLoaderBase';
import * as i0 from "@angular/core";
export class ThCallbackLoaderService extends ThLoader {
    load(...args) {
        const loader = this.createLoaderInstance();
        return loader.load(...args);
    }
}
export class ThCallbackLoaderBasePipe {
    transform(...args) {
        return this.service.load(...args);
    }
}
export class ThCallbackLoaderBaseDirective {
    set url(url) {
        this._url = url;
        this.load();
    }
    get url() {
        return this._url;
    }
    get onLoaded() {
        if (!this.onLoaded$) {
            this.onLoaded$ = new EventEmitter();
        }
        return this.onLoaded$;
    }
    get onProgress() {
        if (!this.onProgress$) {
            this.onProgress$ = new EventEmitter();
        }
        return this.onProgress$;
    }
    constructor(host, zone) {
        this.host = host;
        this.zone = zone;
        this.initialized = false;
    }
    ngOnInit() {
        this.initialized = true;
        this.load();
        // this.zone.runOutsideAngular(() => );
    }
    async load() {
        if (!this.initialized) {
            return;
        }
        if (this._url === undefined) {
            throw new Error('missing url');
        }
        const url = this._url;
        const onProgress = isObserved(this.onProgress$)
            ? (progress) => {
                this.onProgress$?.next(progress);
            }
            : undefined;
        const onLoad = isObserved(this.onLoaded$)
            ? (...args) => {
                this.onLoaded$?.next(args[0]);
            }
            : undefined;
        this.host.objRef = this.zone.runOutsideAngular(() => this.service.load(url, onProgress, onLoad));
    }
}
ThCallbackLoaderBaseDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCallbackLoaderBaseDirective, deps: "invalid", target: i0.ɵɵFactoryTarget.Directive });
ThCallbackLoaderBaseDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThCallbackLoaderBaseDirective, inputs: { url: "url" }, outputs: { onLoaded: "onLoaded", onProgress: "onProgress" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCallbackLoaderBaseDirective, decorators: [{
            type: Directive,
            args: [{}]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }]; }, propDecorators: { url: [{
                type: Input
            }], onLoaded: [{
                type: Output
            }], onProgress: [{
                type: Output
            }] } });
