// ngx-three/esm2020/lib/loaders/ThCubeTextureLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { CubeTextureLoader } from 'three';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from './ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThTexture";
export class CubeTextureLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = CubeTextureLoader;
    }
}
CubeTextureLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: CubeTextureLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
CubeTextureLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: CubeTextureLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: CubeTextureLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThCubeTextureLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThCubeTextureLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTextureLoaderPipe, deps: [{ token: CubeTextureLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThCubeTextureLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTextureLoaderPipe, name: "loadCubeTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTextureLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadCubeTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: CubeTextureLoaderService }]; } });
export class ThCubeTextureLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThCubeTextureLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTextureLoaderDirective, deps: [{ token: i1.ThTexture, host: true }, { token: i0.NgZone }, { token: CubeTextureLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThCubeTextureLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThCubeTextureLoaderDirective, selector: "[loadCubeTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThCubeTextureLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadCubeTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: CubeTextureLoaderService }]; } });
