// ngx-three/esm2020/lib/loaders/ThFBXLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { FBXLoader } from 'three/examples/jsm/loaders/FBXLoader';
import { ThAsyncLoaderBaseDirective, ThAsyncLoaderBasePipe, ThAsyncLoaderService } from './ThAsyncLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThObject3D";
export class FBXLoaderService extends ThAsyncLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = FBXLoader;
    }
}
FBXLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: FBXLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
FBXLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: FBXLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: FBXLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThFBXLoaderPipe extends ThAsyncLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThFBXLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThFBXLoaderPipe, deps: [{ token: FBXLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThFBXLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThFBXLoaderPipe, name: "loadFBX" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThFBXLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadFBX',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: FBXLoaderService }]; } });
export class ThFBXLoaderDirective extends ThAsyncLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
    getRefFromResponse(response) {
        return response;
    }
}
ThFBXLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThFBXLoaderDirective, deps: [{ token: i1.ThObject3D, host: true }, { token: i0.NgZone }, { token: FBXLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThFBXLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThFBXLoaderDirective, selector: "[loadFBX]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThFBXLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadFBX]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: FBXLoaderService }]; } });
