// ngx-three/esm2020/lib/loaders/ThGLTFLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader';
import { ThAsyncLoaderBaseDirective, ThAsyncLoaderBasePipe, ThAsyncLoaderService } from './ThAsyncLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "./compressed-texture/ThDRACOLoader";
import * as i2 from "../generated/ThObject3D";
export class GLTFLoaderService extends ThAsyncLoaderService {
    constructor(dracoLoaderService) {
        super();
        this.dracoLoaderService = dracoLoaderService;
        this.clazz = GLTFLoader;
    }
    createLoaderInstance() {
        const loader = super.createLoaderInstance();
        loader.setDRACOLoader(this.dracoLoaderService.createLoaderInstance());
        return loader;
    }
}
GLTFLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: GLTFLoaderService, deps: [{ token: i1.DRACOLoaderService }], target: i0.ɵɵFactoryTarget.Injectable });
GLTFLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: GLTFLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: GLTFLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.DRACOLoaderService }]; } });
export class ThGLTFLoaderPipe extends ThAsyncLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThGLTFLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGLTFLoaderPipe, deps: [{ token: GLTFLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThGLTFLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThGLTFLoaderPipe, name: "loadGLTF" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGLTFLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadGLTF',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: GLTFLoaderService }]; } });
export class ThGLTFLoaderDirective extends ThAsyncLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
    getRefFromResponse(response) {
        return response.scene;
    }
}
ThGLTFLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGLTFLoaderDirective, deps: [{ token: i2.ThObject3D, host: true }, { token: i0.NgZone }, { token: GLTFLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThGLTFLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThGLTFLoaderDirective, selector: "[loadGLTF]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThGLTFLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadGLTF]'
                }]
        }], ctorParameters: function () { return [{ type: i2.ThObject3D, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: GLTFLoaderService }]; } });
