// ngx-three/esm2020/lib/loaders/ThOBJLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { OBJLoader } from 'three/examples/jsm/loaders/OBJLoader';
import { ThAsyncLoaderBaseDirective, ThAsyncLoaderBasePipe, ThAsyncLoaderService } from './ThAsyncLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThObject3D";
export class OBJLoaderService extends ThAsyncLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = OBJLoader;
    }
}
OBJLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: OBJLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
OBJLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: OBJLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: OBJLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThObjLoaderPipe extends ThAsyncLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThObjLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThObjLoaderPipe, deps: [{ token: OBJLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThObjLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThObjLoaderPipe, name: "loadObj" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThObjLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadObj',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: OBJLoaderService }]; } });
export class ThObjLoaderDirective extends ThAsyncLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
    getRefFromResponse(response) {
        return response;
    }
}
ThObjLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThObjLoaderDirective, deps: [{ token: i1.ThObject3D, host: true }, { token: i0.NgZone }, { token: OBJLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThObjLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThObjLoaderDirective, selector: "[loadObj]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThObjLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadObj]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThObject3D, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: OBJLoaderService }]; } });
