// ngx-three/esm2020/lib/loaders/ThPLYLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { PLYLoader } from 'three/examples/jsm/loaders/PLYLoader';
import { ThAsyncLoaderBaseDirective, ThAsyncLoaderBasePipe, ThAsyncLoaderService } from './ThAsyncLoaderBase';
import { isObserved } from '../util';
import * as i0 from "@angular/core";
import * as i1 from "../generated";
export class PLYLoaderService extends ThAsyncLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = PLYLoader;
    }
}
PLYLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: PLYLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
PLYLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: PLYLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: PLYLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThPLYLoaderPipe extends ThAsyncLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThPLYLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPLYLoaderPipe, deps: [{ token: PLYLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThPLYLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThPLYLoaderPipe, name: "loadPLY" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPLYLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadPLY',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: PLYLoaderService }]; } });
export class ThPLYLoaderDirective extends ThAsyncLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
    getRefFromResponse(response) {
        response.computeVertexNormals();
        return response;
    }
    async load() {
        if (!this.initialized) {
            return;
        }
        if (this._url === undefined) {
            throw new Error('missing url');
        }
        const url = this._url;
        const onProgress = isObserved(this.onProgress$)
            ? (progress) => {
                this.onProgress$?.next(progress);
            }
            : undefined;
        const result = await this.zone.runOutsideAngular(() => this.service.load(url, onProgress));
        // add the new object to the parent and
        // emit a loaded event directly on the three.js object and on objRef$
        this.host.objRef = this.getRefFromResponse(result);
        if (this.onLoaded$ && result !== undefined) {
            this.onLoaded$?.next(result);
        }
    }
}
ThPLYLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPLYLoaderDirective, deps: [{ token: i1.ThBufferGeometry, host: true }, { token: i0.NgZone }, { token: PLYLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThPLYLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThPLYLoaderDirective, selector: "[loadPLY]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThPLYLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadPLY]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThBufferGeometry, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: PLYLoaderService }]; } });
