// ngx-three/esm2020/lib/loaders/ThTextureLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { TextureLoader } from 'three';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from './ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../generated/ThTexture";
export class TextureLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = TextureLoader;
    }
}
TextureLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: TextureLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
TextureLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: TextureLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: TextureLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThTextureLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThTextureLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTextureLoaderPipe, deps: [{ token: TextureLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThTextureLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThTextureLoaderPipe, name: "loadTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTextureLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: TextureLoaderService }]; } });
export class ThTextureLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThTextureLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTextureLoaderDirective, deps: [{ token: i1.ThTexture, host: true }, { token: i0.NgZone }, { token: TextureLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThTextureLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThTextureLoaderDirective, selector: "[loadTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTextureLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: TextureLoaderService }]; } });
