// ngx-three/esm2020/lib/renderer/th-animation-loop.service.mjs
import { Injectable } from '@angular/core';
import { Subject, takeUntil } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../ThEngine.service";
export class ThAnimationLoopService {
    constructor(engineService, ngZone) {
        this.engineService = engineService;
        this.ngZone = ngZone;
        this.active = true;
        this.destroyed$ = new Subject();
        this.beforeRender$ = this.engineService.beforeRender$;
        this.engineService.resize$.pipe(takeUntil(this.destroyed$)).subscribe(() => this.requestAnimationFrame());
    }
    ngOnDestroy() {
        this.stop();
        this.destroyed$.next();
        this.destroyed$.complete();
    }
    /**
     * start the loop
     */
    start() {
        if (!this.active) {
            this.active = true;
            this.requestAnimationFrame();
        }
    }
    /**
     * stop the loop
     */
    stop() {
        this.active = false;
        if (this.frameId !== undefined) {
            cancelAnimationFrame(this.frameId);
            this.frameId = undefined;
        }
    }
    /**
     *
     * @returns true if the render loop is active
     */
    isActive() {
        return this.active;
    }
    /**
     * trigger an animation frame request
     */
    requestAnimationFrame() {
        if (this.frameId === undefined) {
            this.ngZone.runOutsideAngular(() => (this.frameId = requestAnimationFrame(() => {
                this.frameId = undefined;
                this.engineService.render();
                if (this.active) {
                    this.requestAnimationFrame();
                }
            })));
        }
    }
}
ThAnimationLoopService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAnimationLoopService, deps: [{ token: i1.ThEngineService }, { token: i0.NgZone }], target: i0.ɵɵFactoryTarget.Injectable });
ThAnimationLoopService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAnimationLoopService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThAnimationLoopService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ThEngineService }, { type: i0.NgZone }]; } });
