// ngx-three/esm2020/lib/renderer/th-render.directive.mjs
import { Directive, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../ThEngine.service";
import * as i2 from "./th-animation-loop.service";
export class ThRenderDirective {
    constructor(engineService, animationLoopService) {
        this.engineService = engineService;
        this.animationLoopService = animationLoopService;
    }
    get beforeRender() {
        return this.engineService.beforeRender$;
    }
    get onResize() {
        return this.engineService.resize$;
    }
    set renderOnDemand(onDemand) {
        if (onDemand) {
            this.animationLoopService.stop();
        }
        else {
            this.animationLoopService.start();
        }
    }
    get renderOnDemand() {
        return !this.animationLoopService.isActive();
    }
}
ThRenderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRenderDirective, deps: [{ token: i1.ThEngineService }, { token: i2.ThAnimationLoopService }], target: i0.ɵɵFactoryTarget.Directive });
ThRenderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThRenderDirective, selector: "[beforeRender], [onResize], [renderOnDemand]", inputs: { renderOnDemand: "renderOnDemand" }, outputs: { beforeRender: "beforeRender", onResize: "onResize" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRenderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[beforeRender], [onResize], [renderOnDemand]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThEngineService }, { type: i2.ThAnimationLoopService }]; }, propDecorators: { beforeRender: [{
                type: Output
            }], onResize: [{
                type: Output
            }], renderOnDemand: [{
                type: Input
            }] } });
