// ngx-three/esm2020/lib/loaders/data-texture/ThEXRLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { EXRLoader } from 'three/examples/jsm/loaders/EXRLoader';
import { ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe, ThCallbackLoaderService } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThDataTexture";
export class EXRLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = EXRLoader;
    }
}
EXRLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: EXRLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
EXRLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: EXRLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: EXRLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThEXRLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThEXRLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThEXRLoaderPipe, deps: [{ token: EXRLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThEXRLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThEXRLoaderPipe, name: "loadEXRTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThEXRLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadEXRTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: EXRLoaderService }]; } });
export class ThEXRLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThEXRLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThEXRLoaderDirective, deps: [{ token: i1.ThDataTexture, host: true }, { token: i0.NgZone }, { token: EXRLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThEXRLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThEXRLoaderDirective, selector: "[loadEXRTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThEXRLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadEXRTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThDataTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: EXRLoaderService }]; } });
