// ngx-three/esm2020/lib/loaders/data-texture/ThLogLuvLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { LogLuvLoader } from 'three/examples/jsm/loaders/LogLuvLoader';
import { ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe, ThCallbackLoaderService } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThDataTexture";
export class LogLuvLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = LogLuvLoader;
    }
}
LogLuvLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: LogLuvLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
LogLuvLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: LogLuvLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: LogLuvLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThLogLuvLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThLogLuvLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLogLuvLoaderPipe, deps: [{ token: LogLuvLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThLogLuvLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThLogLuvLoaderPipe, name: "loadLogLuvTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLogLuvLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadLogLuvTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: LogLuvLoaderService }]; } });
export class ThLogLuvLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThLogLuvLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLogLuvLoaderDirective, deps: [{ token: i1.ThDataTexture, host: true }, { token: i0.NgZone }, { token: LogLuvLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThLogLuvLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThLogLuvLoaderDirective, selector: "[loadLogLuvTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThLogLuvLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadLogLuvTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThDataTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: LogLuvLoaderService }]; } });
