// ngx-three/esm2020/lib/loaders/data-texture/ThRGBMLoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { RGBMLoader } from 'three/examples/jsm/loaders/RGBMLoader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThDataTexture";
export class RGBMLoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = RGBMLoader;
    }
}
RGBMLoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RGBMLoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
RGBMLoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RGBMLoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RGBMLoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThRGBMLoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThRGBMLoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBMLoaderPipe, deps: [{ token: RGBMLoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThRGBMLoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThRGBMLoaderPipe, name: "loadRGBMTexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBMLoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadRGBMTexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: RGBMLoaderService }]; } });
export class ThRGBMLoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThRGBMLoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBMLoaderDirective, deps: [{ token: i1.ThDataTexture, host: true }, { token: i0.NgZone }, { token: RGBMLoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThRGBMLoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThRGBMLoaderDirective, selector: "[loadRGBMTexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBMLoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadRGBMTexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThDataTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: RGBMLoaderService }]; } });
