// ngx-three/esm2020/lib/loaders/data-texture/ThRGBELoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { RGBELoader } from 'three/examples/jsm/loaders/RGBELoader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThDataTexture";
export class RGBELoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = RGBELoader;
    }
}
RGBELoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RGBELoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
RGBELoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RGBELoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: RGBELoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThRGBELoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThRGBELoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBELoaderPipe, deps: [{ token: RGBELoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThRGBELoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThRGBELoaderPipe, name: "loadRGBETexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBELoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadRGBETexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: RGBELoaderService }]; } });
export class ThRGBELoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThRGBELoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBELoaderDirective, deps: [{ token: i1.ThDataTexture, host: true }, { token: i0.NgZone }, { token: RGBELoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThRGBELoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThRGBELoaderDirective, selector: "[loadRGBETexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThRGBELoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadRGBETexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThDataTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: RGBELoaderService }]; } });
