// ngx-three/esm2020/lib/loaders/data-texture/ThTGALoader.mjs
import { Directive, Host, Injectable, Pipe } from '@angular/core';
import { TGALoader } from 'three/examples/jsm/loaders/TGALoader';
import { ThCallbackLoaderService, ThCallbackLoaderBaseDirective, ThCallbackLoaderBasePipe } from '../ThCallbackLoaderBase';
import * as i0 from "@angular/core";
import * as i1 from "../../generated/ThDataTexture";
export class TGALoaderService extends ThCallbackLoaderService {
    constructor() {
        super(...arguments);
        this.clazz = TGALoader;
    }
}
TGALoaderService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: TGALoaderService, deps: null, target: i0.ɵɵFactoryTarget.Injectable });
TGALoaderService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: TGALoaderService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: TGALoaderService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
export class ThTGALoaderPipe extends ThCallbackLoaderBasePipe {
    constructor(service) {
        super();
        this.service = service;
    }
}
ThTGALoaderPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTGALoaderPipe, deps: [{ token: TGALoaderService }], target: i0.ɵɵFactoryTarget.Pipe });
ThTGALoaderPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "15.1.1", ngImport: i0, type: ThTGALoaderPipe, name: "loadTGATexture" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTGALoaderPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'loadTGATexture',
                    pure: true
                }]
        }], ctorParameters: function () { return [{ type: TGALoaderService }]; } });
export class ThTGALoaderDirective extends ThCallbackLoaderBaseDirective {
    constructor(host, zone, service) {
        super(host, zone);
        this.host = host;
        this.zone = zone;
        this.service = service;
    }
}
ThTGALoaderDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTGALoaderDirective, deps: [{ token: i1.ThDataTexture, host: true }, { token: i0.NgZone }, { token: TGALoaderService }], target: i0.ɵɵFactoryTarget.Directive });
ThTGALoaderDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "15.1.1", type: ThTGALoaderDirective, selector: "[loadTGATexture]", usesInheritance: true, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.1.1", ngImport: i0, type: ThTGALoaderDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[loadTGATexture]'
                }]
        }], ctorParameters: function () { return [{ type: i1.ThDataTexture, decorators: [{
                    type: Host
                }] }, { type: i0.NgZone }, { type: TGALoaderService }]; } });
